src/transaction.js

```diff
@@ -304,6 +304,9 @@ Transaction.prototype.toBuffer = function () {
 
   writeUInt32(this.locktime)
 
+  // avoid slicing unless necessary
+  if (initialOffset !== undefined) return buffer.slice(initialOffset, offset)
```

Comment:
I don't understand, for what purpose are you using `slice` here?
In browser, `buffer` uses `subarray` for `slice`, i.e. if you change sliced array -- parent array also will be changed.